chain/client-primitives/src/types.rs

```diff
@@ -150,14 +150,41 @@ impl SyncStatus {
 /// Actor message requesting block by id or hash.
 pub struct GetBlock(pub BlockReference);
 
+#[derive(thiserror::Error, Debug)]
+pub enum GetBlockError {
+    #[error("IO Error: {0}")]
+    IOError(String),
+    #[error("Block `{0}` is missing")]
+    BlockMissing(CryptoHash),
+    #[error("Block not found")]
+    BlockNotFound(String),
+    #[error("There are no fully synchronized blocks yet")]
+    NotSyncedYet,
+    #[error("Unexpected error occurred: {0}")]
+    Unexpected(String),
```

Comment:
Why do we need `Unexpected`?

Reply:
@khorolets let's rename it to `Unreachable` as well and add the comment as we did for the `RpcBlockError::Unreachable`.